src/testing/flow_solver.h

```diff
@@ -0,0 +1,53 @@
+#ifndef __FLOW_SOLVER_H__
+#define __FLOW_SOLVER_H__
+
+// for FlowSolver class:
+#include "initial_condition.h"
+#include "tests.h"
+#include "dg/dg.h"
+#include "physics/physics.h"
+#include "parameters/all_parameters.h"
+
+// for generate_grid
+#include <deal.II/distributed/shared_tria.h>
+#include <deal.II/distributed/tria.h>
+
+namespace PHiLiP {
+namespace Tests {
+/// Selects which flow case to simulate.
+template <int dim, int nstate>
+class FlowSolver: public TestsBase
+{
+public:
+    /// Constructor.
+    FlowSolver(const Parameters::AllParameters *const parameters_input);
+    
+    /// Destructor
+    ~FlowSolver() {};
+
+    std::shared_ptr< InitialConditionFunction_FlowSolver<dim,double> > initial_condition_function; ///< Initial condition function
+    double domain_left; ///< Domain left-boundary value for generating the grid
+    double domain_right; ///< Domain right-boundary value for generating the grid
+    double domain_volume; ///< Domain volume
+    bool is_taylor_green_vortex = false; ///< Identifies if taylor green vortex case; initialized as false.
+    bool is_triply_periodic_cube = false; ///< Identifies if grid is a triply periodic cube; initialized as false.
```

Comment:
Seems TGV specific. If you want a general class, you'd want to design without any mentions or concepts of TGV.

Reply:
This has been moved to the derived class `PeriodicCubeFlow`